### ISSUE TITLE
Google chrome freezes on form fields

### ISSUE THREAD

**Issue by hozosch** [nvaccess/nvda#9733]
### Steps to reproduce:
Open a sign in site in chrome, eg. on git hub
### Actual behavior:
NVDA and Chrome both freeze
### Expected behavior:
It works as usual
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2019.1.1
#### Windows version:
6.1.7601 Service Pack 1 Build 7601
#### Name and version of other software in use when reproducing the issue:
Google Chrome 75.0.3770.80
#### Does the issue still occur after restarting your PC?
Yes. The interesting thing is that it only does it very recently, it didn't do it in the past.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, the latest alpha snapshot does the same on my computer.

**Comment by Adriani90** [comment]
@hozosch I am testing with NVDA alpha-17611,7b2861fc and Chrome 75, I cannot reproduce this issue. I was online on google crhome in the last week and could never reproduce this. Are you still having this issue? Could it be related to an addon installed? Try to reproduce this issue with addons disabled in NVDA, and then disable addons in Chrome and try again. Is it still occuring?

**Comment by hozosch** [comment]
I can still reproduce it, and neither disabling addons in NVDA nor doing it in chrome change a thing.

**Comment by hozosch** [comment]
I also tried reinstalling chrome completely yesterday and that also didn't work.

**Comment by Adriani90** [comment]
Hmm that’S strange. Could you please run the com registration fixing tool in nvda menu / tools? If this does not work, I supose there could be some temporary data, coockies etc. which slow down the program. Maybe you should try to run a clean process on your windows partition in the properties dialog and remove all temporary data including coockies etc.

 

 

 

Von: hozosch <notifications@github.com> 
Gesendet: Freitag, 14. Juni 2019 22:30
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Comment <comment@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Google chrome freezes on form fields (#9733)

 

I also tried reinstalling chrome completely yesterday and that also didn't work.

**Comment by hozosch** [comment]
Tried everything you said, nothing works. I also ran c cleaner over it, no success.

**Comment by JulienCochuyt** [comment]
Reproduced here, and it is very annoying.

Windows 10 Professional version 1809 (17763.557) 64 bits with French locale.
Google Chrome version 75.0.3770.90 (64 bits)

NVDA versions:
 - 2016.4 portable
 - 2018.1 portable
 - 2019.1.1 portable
 - 2019.2beta1 installed
 - alpha-17611,7b2861fc portable

I first ran the COM Registration Fixing Tool of 2019.2beta1, then restarted my system, and could reproduce the issue with all the above listed NVDA versions, without any addon and fresh clean configs, restarting the system between each test to be sure.
Chrome freezes and waiting for a dozen minutes does not change anything.
NVDA is not responsive within the frozen Chrome, but is still functional once focusing any other application.

@Adriani90, could you please give the full versions of Google Chrome and Windows you did not reproduce with?

GitHub sign-in page is a good example for this issue, among many, many others.
I could not yet write a short HTML snippet to reproduce.
It seems a form with a password input is not enough.
I'll try harder to narrow down the simpler scenario to reproduce as this issue is really a show-stopper here.

**Comment by JulienCochuyt** [comment]
Could not reproduce right now on GitHub sign-in page. Maybe they have changed something as they roll out updates so often.
I have tested a few dozen other random sign-in pages without issue.
The faulty conditions do not seem to be so easily met. Still, they are met on critical business apps here.

Example page to reproduce: https://www.odoo.com/web/login

**Comment by JulienCochuyt** [comment]
On my side, the issue seems to only reproduce when I have a login/password stored in Chrome for the tested page. If I remove the stored credentials, the page works just fine. Once stored again, loading the page in the current tab freezes Chrome again.

**Comment by Adriani90** [comment]
@JulienCochuyt thanks for this note. I confirm this issue now as follows:
1. Open Github in Chrome,
2. Make sure your credentials for github are stored
3. Go to sign in
4. In the user name edit field, press the first letter of your user name and press down arrow
5. Press tab to go to the password field

Actual: Chrome crashes
Expected: Chrome should not crash.

**Comment by Adriani90** [comment]
Following is written to the log:

```
IO - inputCore.InputManager.executeGesture (11:37:22.197):
Input: kb(laptop):a
IO - inputCore.logTimeSinceInput (11:37:22.247):
0.051 sec since input
IO - speech.speak (11:37:22.247):
Speaking [u'mit AutoVervollst\xe4ndigung']
IO - inputCore.InputManager.executeGesture (11:37:23.342):
Input: kb(laptop):downArrow
DEBUG - editableText.EditableText._hasCaretMoved (11:37:23.401):
Focus event. Elapsed: 0 ms
IO - inputCore.InputManager.executeGesture (11:37:24.589):
Input: kb(laptop):tab
IO - inputCore.InputManager.executeGesture (11:37:25.204):
Input: kb(laptop):tab
IO - inputCore.InputManager.executeGesture (11:37:29.852):
Input: kb(laptop):shift+tab
IO - inputCore.InputManager.executeGesture (11:37:31.512):
Input: kb(laptop):windows+m
DEBUGWARNING - watchdog._watcher (11:37:33.434):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 231, in <module>
  File "core.pyc", line 519, in main
  File "wx\core.pyc", line 2134, in MainLoop
  File "gui\__init__.pyc", line 1003, in Notify
  File "core.pyc", line 490, in run
  File "queueHandler.pyc", line 86, in pumpAll
  File "queueHandler.pyc", line 53, in flushQueue
  File "scriptHandler.pyc", line 150, in _queueScriptCallback
  File "scriptHandler.pyc", line 192, in executeScript
  File "editableText.pyc", line 185, in script_caret_moveByLine
  File "editableText.pyc", line 144, in _caretMovementScriptHelper
  File "editableText.pyc", line 129, in _caretScriptPostMovedHelper
  File "NVDAObjects\IAccessible\ia2TextMozilla.pyc", line 471, in expand
  File "NVDAObjects\IAccessible\ia2TextMozilla.pyc", line 355, in _findUnitEndpoints
  File "NVDAObjects\IAccessible\__init__.pyc", line 161, in expand
  File "textInfos\offsets.pyc", line 458, in expand
  File "textInfos\offsets.pyc", line 437, in _getUnitOffsets
  File "NVDAObjects\IAccessible\__init__.pyc", line 277, in _getLineOffsets
  File "comtypes\__init__.pyc", line 857, in __call__

DEBUGWARNING - scriptHandler.executeScript (11:37:33.486):
error executing script: <bound method Dynamic_EditableTextWithAutoSelectDetectionEditorIa2WebIAccessible.script_caret_moveByLine of <NVDAObjects.Dynamic_EditableTextWithAutoSelectDetectionEditorIa2WebIAccessible object at 0x053D4090>> with gesture u'Pfeiltaste nach unten'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 192, in executeScript
  File "editableText.pyc", line 185, in script_caret_moveByLine
  File "editableText.pyc", line 144, in _caretMovementScriptHelper
  File "editableText.pyc", line 129, in _caretScriptPostMovedHelper
  File "NVDAObjects\IAccessible\ia2TextMozilla.pyc", line 471, in expand
  File "NVDAObjects\IAccessible\ia2TextMozilla.pyc", line 355, in _findUnitEndpoints
  File "NVDAObjects\IAccessible\__init__.pyc", line 161, in expand
  File "textInfos\offsets.pyc", line 458, in expand
  File "textInfos\offsets.pyc", line 437, in _getUnitOffsets
  File "watchdog.pyc", line 202, in _COMError_init
CallCancelled
DEBUGWARNING - watchdog._watcher (11:37:34.000):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 231, in <module>
  File "core.pyc", line 519, in main
  File "wx\core.pyc", line 2134, in MainLoop
  File "gui\__init__.pyc", line 1003, in Notify
  File "core.pyc", line 490, in run
  File "queueHandler.pyc", line 86, in pumpAll
  File "queueHandler.pyc", line 53, in flushQueue
  File "eventHandler.pyc", line 62, in _queueEventCallback
  File "eventHandler.pyc", line 150, in executeEvent
  File "eventHandler.pyc", line 163, in doPreGainFocus
  File "api.pyc", line 107, in setFocusObject
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 501, in _get_container
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyc", line 955, in _get_parent
  File "IAccessibleHandler.pyc", line 434, in accParent

DEBUGWARNING - watchdog._watcher (11:37:34.555):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 231, in <module>
  File "core.pyc", line 519, in main
  File "wx\core.pyc", line 2134, in MainLoop
  File "gui\__init__.pyc", line 1003, in Notify
  File "core.pyc", line 490, in run
  File "queueHandler.pyc", line 86, in pumpAll
  File "queueHandler.pyc", line 53, in flushQueue
  File "eventHandler.pyc", line 62, in _queueEventCallback
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\IAccessible\__init__.pyc", line 1845, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1030, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 918, in reportFocus
  File "speech.pyc", line 367, in speakObject
  File "baseObject.pyc", line 31, in __get__
  File "NVDAObjects\IAccessible\__init__.pyc", line 820, in _get_role
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyc", line 805, in _get_IAccessibleRole

DEBUGWARNING - eventHandler.executeEvent (11:37:34.608):
error executing event: gainFocus on <NVDAObjects.Dynamic_MenuItemIa2WebIAccessible object at 0x0588F5F0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 155, in executeEvent
  File "eventHandler.pyc", line 92, in __init__
  File "eventHandler.pyc", line 100, in next
  File "NVDAObjects\IAccessible\__init__.pyc", line 1845, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 1030, in event_gainFocus
  File "NVDAObjects\__init__.pyc", line 918, in reportFocus
  File "speech.pyc", line 367, in speakObject
  File "baseObject.pyc", line 31, in __get__
  File "NVDAObjects\IAccessible\__init__.pyc", line 820, in _get_role
  File "baseObject.pyc", line 47, in __get__
  File "baseObject.pyc", line 147, in _getPropertyViaCache
  File "watchdog.pyc", line 202, in _COMError_init
CallCancelled
IO - inputCore.InputManager.executeGesture (11:37:34.825):
Input: kb(laptop):windows+m
```

**Comment by Adriani90** [comment]
This issue does not occur with Narator.
cc: @michaelDCurran, @feerrenrut

**Comment by JulienCochuyt** [comment]
I have just tested a few different versions of Google Chrome.
I can only reproduce with 75.0.3770.80 and 75.0.3770.90
I cannot reproduce with versions prior or equal to 74.0.3729.169

Given that 75.0.3770.80 rolled out on 2019-06-04 and updates are automatic, I *suspect* this version was already installed and working fine on my system last week, and that it might have broken in conjunction with latest Windows 10 update 1809 (17763.557). This is only a suspicion, no evidence.

**Comment by JulienCochuyt** [comment]
Google Chrome update to 75.0.3770.100 thankfully fixes the issue on my side.

@hozosch, do you confirm?

**Comment by Adriani90** [comment]
I confirm in Chrome 75.0.3770.100 this issue does not occur. I am closing this as works for me. @hozosch please comment on it if you are still having the problem and we can reopen the discussion. Thanks.
